crowbar_framework/lib/crowbar/upgrade_status.rb

```diff
@@ -97,7 +97,7 @@ def finished?
 
     def save
       progress_file_path.open("w") do |f|
-        f.write(JSON.pretty_generate(progress))
+        f.write(YAML.dump(progress))
```

Comment:
I extracted the commit here https://github.com/crowbar/crowbar-core/pull/769. I believe we need this before the thread safe locking is done to finally make use of the progress library and unblock UI and QA.